### ISSUE TITLE
Build issue on windows using GTX 1080

### ISSUE THREAD

**Issue by lasandcris** [BVLC/caffe#4426]
In the windows readme, it states that cuDNN 3 and 4 should be used to build caffe. However, the GTX 1080 is using Cuda 8 and needs cuDNN 5. When can we see an updated caffe windows branch to support it?

thanks

**Comment by seanbell** [comment]
My understanding is that the windows branch is currently not being worked on and needs a new maintainer(s).  As a workaround in the meantime, you could either run without cuDNN, or use linux.

**Comment by lasandcris** [comment]
@seanbell , thanks Sean. That's what I did - compiled without cuDNN - it works without any problem.

**Comment by msly** [comment]
compiled without cuDNN means onlyCPU?

**Comment by seanbell** [comment]
@msly no, that is a separate flag.  Caffe has built-in support for the GPU without cuDNN, it's just slower.

**Comment by ValleZ** [comment]
@seanbell mind to share steps to do the same?

**Comment by lasandcris** [comment]
@ValleZ - Just build without cuDNN and use CUDA8. I don't know how much faster it is with cuDNN but its still very fast.

**Comment by lasandcris** [comment]
@ValleZ - as for CUDA architecture, I used:
compute_35,sm_35;compute_60,sm_60 
which works well enough. But if someone has a better tried and tested setting please let us know. 
thanks
_laszlo

**Comment by msly** [comment]
I used compute_52,sm_52;compute_61,sm_61

https://developer.nvidia.com/cuda-gpus

**Comment by lasandcris** [comment]
@msly - I did a quick comparison against your setting. First, I ran the test_all which came back 12 seconds faster on the settings I used. But then I tested it with a basic classification dataset (using 4800 imgs (256x256 color) for training and 1600 for validation - using the default GooLeNet - 30 epoch - lr 0.01) and it seemed to perform the same in speed (only 2 seconds faster with your settings over a 30 min training). Also, the Loss seemed to have performed slightly better with yours which I found interesting.

**Comment by johnyboyoh** [comment]
@lasandcris Hi, I am also trying to build caffe for windows to work with GTX 1080. I wonder which windows branch did you use ? I am currently using the branch maintained by Microsoft (i.e. https://github.com/Microsoft/caffe?files=1). Is this the branch you used? what changes did you have to do when changing from cuda 7.5 to 8? Many thanks!

**Comment by lasandcris** [comment]
@johnyboyoh - Well, I'm using my own merge of 0.15 Linux branch with the Microsoft one as I needed some other functions. But maybe that would just add to your problems. To answer your question, and to keep it as simple as possible, after you installed cuda8, just go to your installation directory (somewhere like C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA) and you'll see the different versions of cuda installed here. Just rename the version 8 to 7.5 (and if you have the 7.5 installed as well just rename that to something else) and rebuild your solution. (make sure your environment variable is pointing to the right one as well - so if you only installed version 8 then you will have to change the path in your env settings - but if you install both then you don't need to do anything). There is also another way to do this by going into the configs of your solution and fix it that way but that might complicate things. 

Also, don't forget to set your cuDNN to false before building. I'm also working on this to fix it and as soon as I do I'll post the solution. As for the compute settings, there is already two setups - I kept my own as I have been using it for a while and it didn't cause any issues - but you pick which ever you like.  

good luck
_laszlo

**Comment by johnyboyoh** [comment]
@lasandcris thank you for your comment! I guess what you suggested could have been indeed simpler than what I eventually ended up doing, but still - Success! I used the Microsoft branch (I believe now merged with BVLC) on Windows 10 pro with Visual Studio 2013 Express. I installed Cuda 8 and then on CommonSettings.props I set <CudaVersion>8.0</CudaVersion>
trying to compile resulted in an error saying  cublas_v2.h can not be found. 
looking at the build script I realized that it relies on an environment variable named CUDA_PATH
strangely this variable was not set on my platform. After setting it to
c:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\ and restarting Visual Studio, compilation was successful.
